kafka-rest/src/main/java/io/confluent/kafkarest/KafkaRestConfig.java

```diff
@@ -709,6 +729,24 @@ public Properties getOriginalProperties() {
     return properties;
   }
 
+  public final Map<String, Object> getSchemaRegistryConfigs() {
+    HashMap<String, Object> possibleConfigs = new HashMap<>();
+    possibleConfigs.putAll(originals());
+    possibleConfigs.putAll(originalsWithPrefix("schema.registry.", /* strip= */ true));
+    possibleConfigs.putAll(originalsWithPrefix("client.", /* strip= */ true));
+    possibleConfigs.putAll(originalsWithPrefix("producer.", /* strip= */ true));
+    possibleConfigs.putAll(originalsWithPrefix("consumer.", /* strip= */ true));
```

Comment:
is it the case that previously, you could have different SR related settings for producer / consumer which are now clobbered - effectively making this a breaking change?

if so, how likely is that? (i'm assuming not very, but haven't looked into it deeply).

this way of arriving at the config is pretty sloppy, hopefully in v3, you can constrain this so SR settings must be specified with a `schema.registry.` prefix.

Reply:
rather than silently overwrite, maybe consider failing fast and throw an exception on any conflicts?

Reply:
so yes, two messages: make sure you don't break anything important and fail fast if possible. i won't review this in depth unless you want me to, trust you to get it right.

Reply:
You are right, you could have before the producer and the consumer using different schema registries, so yeah, that makes it breaking. I believe that has never been the intention, but rather an artefact of not checking the configs properly. I also don't think this a feature, but rather a bug. I can't tell how often this happen in the wild, but I can tell you whoever is doing it shouldn't. They will get a warning now. The `consumer.` SR will be preferred over the `producer.` for no special reason.